src/main/groovy/liquibase/harness/base/BaseCompatibilityTest.groovy

```diff
@@ -0,0 +1,225 @@
+        String pathToJsonCheckingSqlFile =
+                getResourceContent("/${basePath}checkingSql/${testInput.change}/${testInput.change}Json.sql")
+        String pathToSqlCheckingSqlFile =
+                getResourceContent("/${basePath}checkingSql/${testInput.change}/${testInput.change}Sql.sql")
+        boolean shouldRunChangeSet
+
+        and: "fail test if expectedResultSet is not provided"
+        shouldRunChangeSet = expectedResultSet != null
+        assert shouldRunChangeSet: "No expectedResultSet for ${testInput.change} against " +
+                "${testInput.database.shortName} ${testInput.database.databaseMajorVersion}." +
+                "${testInput.database.databaseMinorVersion}"
+
+        and: "check database under test is online"
+        def connection = testInput.database.getConnection()
+        shouldRunChangeSet = connection instanceof JdbcConnection
+        assert shouldRunChangeSet: "Database ${testInput.databaseName} ${testInput.version} is offline!"
+
+        and: "execute Liquibase validate command to ensure a chagelog is valid"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("validate", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("validate", argsMap) //TODO: Doesn't work for sql-formatted changelogs. Add issue link
+
+        when: "execute XML, YAML, SQL and JSON formatted changelogs using liquibase update command"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("update", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("update", argsMap)
+
+        and: "execute Liquibase tag command. Tagging last row of DATABASECHANGELOG table"
+        argsMap.remove("changeLogFile")
+        argsMap.put("tag", "test_tag")
+        executeCommandScope("tag", argsMap)
+
+        and: "execute Liquibase history command"
+        executeCommandScope("history", argsMap)
+
+        and: "execute Liquibase status command"
+        argsMap.put("changeLogFile", pathToXmlChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToYamlChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToJsonChangelogFile)
+        executeCommandScope("status", argsMap)
+        argsMap.put("changeLogFile", pathToSqlChangelogFile)
+        executeCommandScope("status", argsMap)
```

Comment:
plz check that `status` returned successful response

Reply:
assert executeCommandScope("status", argsMap).toString().contains("is up to date") ++

Reply:
"is up to date" is a part of response to "status" command in case no changesets has left to delpoy, e.g.:
lbuser@jdbc:postgresql://localhost:5437/lbcat is up to date